bundles/org.openhab.binding.wlanthermo/src/main/java/org/openhab/binding/wlanthermo/internal/api/nano/WlanThermoNanoV1CommandHandler.java

```diff
@@ -0,0 +1,266 @@
+                            return true;
+                        }
+                    case CHANNEL_MIN:
+                        if (command instanceof QuantityType) {
+                            channel.setMin(((QuantityType<?>) command).toUnit(unit).doubleValue());
+                            return true;
+                        }
+                    case CHANNEL_MAX:
+                        if (command instanceof QuantityType) {
+                            channel.setMax(((QuantityType<?>) command).toUnit(unit).doubleValue());
+                            return true;
+                        }
+                    case CHANNEL_ALARM_DEVICE:
+                        if (command instanceof OnOffType) {
+                            BigInteger value;
+                            if (command == OnOffType.ON) {
+                                value = BigInteger.valueOf(channel.getAlarm()).setBit(1);
+                            } else {
+                                value = BigInteger.valueOf(channel.getAlarm()).clearBit(1);
+                            }
+                            channel.setAlarm(value.intValue());
+                            return true;
+                        }
+                    case CHANNEL_ALARM_PUSH:
+                        if (command instanceof OnOffType) {
+                            BigInteger value;
+                            if (command == OnOffType.ON) {
+                                value = BigInteger.valueOf(channel.getAlarm()).setBit(0);
+                            } else {
+                                value = BigInteger.valueOf(channel.getAlarm()).clearBit(0);
+                            }
+                            channel.setAlarm(value.intValue());
+                            return true;
+                        }
+                    case CHANNEL_COLOR_NAME:
+                        if (command instanceof StringType) {
+                            channel.setColor(UtilNano.toHex(((StringType) command).toString()));
+                            return true;
+                        }
+                }
+            }
+        } else if (channelUID.getId().equals(CHANNEL_PITMASTER_1)) {
+            if (data.getPitmaster() != null && data.getPitmaster().getPm() != null
+                    && data.getPitmaster().getPm().size() > 0) {
+                Pm pm = data.getPitmaster().getPm().get(0);
+                switch (channelUID.getIdWithoutGroup()) {
+                    case CHANNEL_PITMASTER_CHANNEL_ID:
+                        pm.setChannel(((QuantityType<?>) command).intValue());
+                        return true;
+                    case CHANNEL_PITMASTER_PIDPROFILE:
+                        pm.setPid(((QuantityType<?>) command).intValue());
+                        return true;
+                    case CHANNEL_PITMASTER_SETPOINT:
+                        pm.setSet(((QuantityType<?>) command).toUnit(unit).doubleValue());
```

Comment:
You should make sure that the units are converted to what you expect them to be. Use `QuantityType.toUnit` to handle the conversion.

Reply:
That already included in all channels, where unit conversion might be required.
The PITMASTER_PIDPROFILE channel however only contains an integer reference to a profile, so no conversion needed.

Reply:
If PITMASTER_PIDPROFILE is just a integer reference then why does it need to be a `QuantityType`?
Same question for CHANNEL_PITMASTER_CHANNEL_ID.

Reply:
Ah, now I understood what you meant. 
Reverted to DecimalType.